contracts/viewer/lib/RebalancingSetTokenViewer.sol

```diff
@@ -0,0 +1,97 @@
+/*
+    Copyright 2019 Set Labs Inc.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+pragma solidity 0.5.7;
+
+
+import { IRebalancingSetToken } from "../../core/interfaces/IRebalancingSetToken.sol";
+import { RebalancingLibrary } from "../../core/lib/RebalancingLibrary.sol";
+
+
+/**
+ * @title RebalancingSetTokenViewer
+ * @author Set Protocol
+ *
+ * Interfaces for fetching multiple RebalancingSetToken state in a single read
+ */
+contract RebalancingSetTokenViewer {
+
+    /*
+     * Fetches all RebalancingSetToken state associated with a rebalance proposal
+     *
+     * @param  _rebalancingSetTokenAddress    Address of RebalancingSetToken contract
+     * @return RebalancingLibrary.State       Current rebalance state on the RebalancingSetToken
+     * @return address[]                      Auction proposal library and next allocation SetToken addresses
+     * @return uint256[]                      Auction time to pivot, start price, and pivot price
+     */
+    function fetchRebalanceProposalStateAsync(
+        address _rebalancingSetTokenAddress
```

Comment:
Can pass in the instance instead: `IRebalancingSetToken _rebalancingSetToken`. Addresses passed in from eth_call are automatically coerced into the instance.

That way in your reads below, you can just call the method directly (e.g. `_rebalancingSetToken.rebalanceState()`)

Reply:
+1